pkg/skaffold/jib/jib.go

```diff
@@ -18,13 +18,101 @@ package jib
 
 import (
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha3"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/pkg/errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
-func GetDependenciesMaven(_ /*workspace*/ string, _ /*a*/ *v1alpha3.JibMavenArtifact) ([]string, error) {
-	return nil, errors.New("jib maven support is unimplemented")
+// GetDependenciesMaven finds the source dependencies for the given jib-maven artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesMaven(workspace string, a *v1alpha3.JibMavenArtifact, isWindows bool) ([]string, error) {
+	executable, subCommand := getCommandMaven(workspace, a, isWindows)
+	return getDependencies(workspace, "pom.xml", executable, subCommand, "jib-maven")
 }
 
-func GetDependenciesGradle(_ /*workspace*/ string, _ /*a*/ *v1alpha3.JibGradleArtifact) ([]string, error) {
-	return nil, errors.New("jib gradle support is unimplemented")
+// GetDependenciesGradle finds the source dependencies for the given jib-gradle artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesGradle(workspace string, a *v1alpha3.JibGradleArtifact, isWindows bool) ([]string, error) {
+	executable, subCommand := getCommandGradle(workspace, a, isWindows)
+	return getDependencies(workspace, "build.gradle", executable, subCommand, "jib-gradle")
+}
+
+func exists(workspace string, filename string) bool {
+	_, err := os.Stat(filepath.Join(workspace, filename))
+	return err == nil
+}
+
+func getDependencies(workspace string, buildFile string, executable string, subCommand []string, artifactName string) ([]string, error) {
+	if !exists(workspace, buildFile) {
+		return nil, errors.Errorf("no %s found", buildFile)
+	}
+
+	cmd := exec.Command(executable, subCommand...)
+	cmd.Dir = workspace
+	stdout, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting %s dependencies", artifactName)
+	}
+
+	return getDepsFromStdout(string(stdout)), nil
+}
+
+const (
+	mavenExecutable     = "mvn"
+	mavenWrapper        = "mvnw"
+	mavenWindowsWrapper = "mvnw.cmd"
+
+	gradleExecutable     = "gradle"
+	gradleWrapper        = "gradlew"
+	gradleWindowsWrapper = "gradlew.bat"
+)
+
+func getCommandMaven(workspace string, a *v1alpha3.JibMavenArtifact, isWindows bool) (executable string, subCommand []string) {
```

Comment:
Can we generalize this and getCommandGradle to take the subcommand as an argument?

Reply:
`getCommand` does take `subCommand` (`defaultSubCommand`) as an argument